### PR TITLE
Publishing images to GitHub container registry

### PR DESCRIPTION
**Pull request by pravinpushkar**
Signed-off-by: Pravin Pushkar <ppushkar@microsoft.com>

# Description

Publishing images to GitHub registry along with Docker Hub

## Issue reference

Please reference the issue this PR will close: #4199 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by pravinpushkar** [comment]
I have tried using `${{ github.actor }}` as username to login to `ghcr.io`.  I ran it in my fork it picks the username.

File: .github/workflows/dapr.yml
Comment (berndverst) [inline .github/workflows/dapr.yml:154]: `github.actor` is the username of the user that initiated the workflow run.

This would differ based on who is triggering the workflow.

Should we not hardcode this is to be dapr / dapr bot, or maybe `github.repository_owner` ? I'm unclear on the correct value, but I don't think it is `github.actor`.

Also the secret we have today in our organizational secret must have the right scope to read / write packages. I am not sure whether it does. We may need to roll the organizational secret - but we also have to double check which user this secret is scoped to.

Thoughts?

**Review by berndverst — CHANGES_REQUESTED** [review]

File: .github/workflows/dapr.yml
Comment (daixiang0) [reply .github/workflows/dapr.yml:154]: @pravinpushkar due to safety concerns, please post a PR in your fork repo, then we can see the result here. Then we can pick one maintainer who has permission to use secrets to finish this job.

/cc @yaron2 @artursouza WDYT?

Comment (pravinpushkar) [reply .github/workflows/dapr.yml:154]: @berndverst I just verified `github.repository_owner` is also working and looks to be a better option than `github.actor`.
By default `secrets.GITHUB_TOKEN` has read/write package scope.

As of now I am changing the username to be `github.repository_owner` and will wait for further feedback.

Comment (pravinpushkar) [reply .github/workflows/dapr.yml]: @daixiang0 I created one PR in my fork (https://github.com/pravinpushkar/dapr/runs/5106183840?check_suite_focus=true) .. but publish did not run I guess because of `if: github.event_name != 'pull_request'` under publish in yml file. I will branch out from this recreate the fork PR.

Comment (pravinpushkar) [reply .github/workflows/dapr.yml]: I ran the workflow in my fork PR but it is failing with this reason - `docker manifest create ***/dapr:edge ***/dapr:edge-linux-amd64 ***/dapr:edge-linux-arm ***/dapr:edge-linux-arm64 ***/dapr:edge-windows-amd64
no such manifest: ***/dapr:edge-linux-amd64`
here is the link for the ran action - https://github.com/pravinpushkar/dapr/runs/5122070859?check_suite_focus=true

@artursouza  Any thoughts on this?

Comment (beiwei30) [reply .github/workflows/dapr.yml]: @pravinpushkar I believe it's caused by `${{ secrets.DOCKER_REGISTRY }}` not configured in your repository.

**Comment by pravinpushkar** [comment]
Packages getting published successfully to my personal github container registry and docker registry.
Take a look here  - 
ghcr  - https://github.com/pravinpushkar?tab=packages
docker - https://hub.docker.com/search?q=pravin7007%2F&type=image

**Review by berndverst — APPROVED** [review]

**Review by shubham1172 — APPROVED** [review]

**Review by artursouza — APPROVED** [review]

**Review by artursouza — APPROVED** [review]

**Pull request merged**
